server/src/main/java/org/elasticsearch/cluster/metadata/OperatorMetadata.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.cluster.DiffableUtils;
+import org.elasticsearch.cluster.SimpleDiffable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Metadata class that contains information about cluster settings/entities set
+ * in an operator mode.
+ *
+ * <p>
+ * These types of settings are read only through the REST API,
+ * and cannot be modified by the end user.
+ */
+public record OperatorMetadata(
+    String namespace,
+    Long version,
+    Map<String, OperatorHandlerMetadata> handlers,
+    OperatorErrorMetadata errorMetadata
+) implements SimpleDiffable<OperatorMetadata>, ToXContentFragment {
+
+    private static final ParseField VERSION = new ParseField("version");
+    private static final ParseField HANDLERS = new ParseField("handlers");
+    private static final ParseField ERRORS_METADATA = new ParseField("errors");
+
+    /**
+     * OperatorMetadata contains information about settings set in operator mode.
+     *
+     * <p>
+     * These settings cannot be updated by the end user and are set outside of the
+     * REST layer, e.g. through file based settings or by plugin/modules.
```

Comment:
I forgot to mention this naming issue. Since all of these classes are about more than operator settings, I think we need better terminology that applies to file based settings and programmatic fixed settings. I'm not sure what that naming should be, but as it is the "operator" terminology will be confusing when we start using this in plugins and modules.

Reply:
Perhaps a name referencing cluster state. Some ideas:

```
unmodifiable cluster state
immutable cluster state
....
```

So then this class might be `ImmutableClusterStateMetadata`. It's not the best, but it demonstrates the idea.

Reply:
Good idea, I've been struggling with terminology on this. I like the immutable prefix, it's really what it is. Operator is just the mode we set them in initially.